engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -645,7 +656,8 @@ namespace dmGameSystem
 
         for(uint32_t i = 0; i < MAX_TEXTURE_COUNT; ++i)
         {
-            ro.m_Textures[i] = GetTexture(component, component->m_Resource, i);
+            TextureResource* texture_res = GetTextureResource(component, component->m_Resource, i);
+            ro.m_Textures[i] = GetTexture(texture_res);
```

Comment:
Imho, this code looks a little verbose, since the GetTexture only really gets the texture, but also has to check the resource first.

Perhaps have a `HTexture GetTexture(component, resource, i)`?